src/main/java/com/google/firebase/cloud/StorageClient.java

```diff
@@ -113,16 +118,16 @@ public String getDownloadUrl(Blob blob) {
     if (metadata == null) {
       metadata = new HashMap<>();
     }
-    String downloadTokens = metadata.get(DOWNLOAD_TOKENS);
+    String downloadTokens = metadata.get(DOWNLOAD_TOKENS_METADATA_KEY);
     if (Strings.isNullOrEmpty(downloadTokens)) {
       downloadTokens = UUID.randomUUID().toString();
-      metadata.put(DOWNLOAD_TOKENS, downloadTokens);
+      metadata.put(DOWNLOAD_TOKENS_METADATA_KEY, downloadTokens);
       blob.toBuilder().setMetadata(metadata).build().update();
     }
 
     String token = downloadTokens.split(",")[0];
     return String.format("https://firebasestorage.googleapis.com/v0/b/%s/o/%s?alt=media&token=%s",
-        blob.getBucket(), blob.getName(), token);
+        blob.getBucket(), blob.getName(), UrlEscapers.urlFormParameterEscaper().escape(token));
```

Comment:
Escaped the object name. Bucket names are URL safe as per https://cloud.google.com/storage/docs/naming#requirements